src/main/java/com/github/manolo8/darkbot/core/manager/HeroManager.java

```diff
@@ -140,18 +141,35 @@ public boolean setMode(Config.ShipConfig config) {
         return setMode(config.CONFIG, config.FORMATION);
     }
 
+    private boolean formationCheckBySlot = true;
     public boolean setMode(int con, Character form) {
         int formationCheck = main.config.GENERAL.FORMATION_CHECK;
 
         if (this.config != con && System.currentTimeMillis() - configTime > 5500L) {
             Main.API.keyboardClick(keybinds.getCharCode(TOGGLE_CONFIG));
             this.configTime = System.currentTimeMillis();
         }
-        boolean checkFormation = formationCheck > 0 && (System.currentTimeMillis() - formationTime) > formationCheck * 1000;
+        boolean checkFormation = formationCheck > 0 && (System.currentTimeMillis() - formationTime) > formationCheck * 1000L;
 
         if ((this.formation != form || checkFormation) && System.currentTimeMillis() - formationTime > 3500L) {
             Main.API.keyboardClick(this.formation = form);
-            if (formation != null) this.formationTime = System.currentTimeMillis();
+            if (formation != null) {
+                this.formationTime = System.currentTimeMillis();
+                this.formationCheckBySlot = false;
+            }
+
+        } else if (!formationCheckBySlot && System.currentTimeMillis() - formationTime > 600L) {
```

Comment:
600 ms is way too little, we should never be clicking to switch formations constantly so quickly

also this will only trigger if it's been in the 3500 ms  timeframe since last formation? does this ever trigger?